src/FixerConfiguration/FixerConfigurationResolver.php

```diff
@@ -66,6 +66,20 @@ public function resolve(array $options)
 
             $allowedValues = $option->getAllowedValues();
             if (null !== $allowedValues) {
+                foreach ($allowedValues as &$allowedValue) {
+                    if ($allowedValue instanceof AllowedValueSubset) {
+                        $allowedValue = function ($values) use ($allowedValue) {
+                            foreach ($values as $value) {
+                                if (!in_array($value, $allowedValue->getValues(), true)) {
+                                    return false;
+                                }
+                            }
+
+                            return true;
+                        };
+                    }
+                }
+
                 $resolver->setAllowedValues($name, $allowedValues);
```

Comment:
Original `setAllowedValues` (`\Symfony\Component\OptionsResolver\OptionsResolver::setAllowedValues`) we rely on here accepts:
> * @param mixed  $allowedValues One or more acceptable values/closures

that mixed is in fact `array<mixed|callable>`, and we were following that so far, as our `FixerOptionBuilder` strongly mimics Sf one.
Now, we extends it to `array<mixed|callable|AllowedValueSubset>` . it could be worth documenting.

but here, it's other thing as well.
our public `FixerOptionBuilder` (non-`@internal`) is now working with `@internal` `AllowedValueSubset`

Reply:
Actually `OptionsResolver::setAllowedValues` only accepts `\Closure` as validator, not any callable.

I updated the PR: now `FixerConfigurationResolver` checks if the allowed value is a callable object (class with an `__invoke()` method) rather than an instance of `AllowedValueSubset`. In such case, it wraps it with a `\Closure`. WDYT?

Reply:
like the idea :+1:

Reply:
like the idea :+1: